### ISSUE TITLE
By default, don't include `data` in linkage if not side-loaded and `link` present 

### ISSUE THREAD

**Issue by samselikoff** [samselikoff/ember-cli-mirage#1146]
https://github.com/samselikoff/ember-cli-mirage/pull/1096 broke some existing behavior due to the way Ember Data handles fetching related resources when both a `data` key and a `links` key is present in a [json:api relationship object](http://jsonapi.org/format/#document-resource-object-relationships).

Previously, we were either using `links` or `data`, but not both. Technically the spec allows both the be present, so #1096 made this change. But this broke existing behavior.

I think the best solution is for Mirage to by default, not include `data` in linkage if `links` are present AND if the related relationship object is not being side-loaded. (Even if you have `links` specified, you'd still want the `data: {}` resource linkage present so Ember Data can cross-reference the `included` models).

Sound correct?

/cc @kellyselden @lancedikson @timhaines

**Comment by timhaines** [comment]
Sounds good to me.

**Comment by lancedikson** [comment]
Yeah, I guess it's ok :)

**Comment by eduardbaitinger** [comment]
> I think the best solution is for Mirage to by default, not include data in linkage if **links are present** AND if the related relationship object is not being side-loaded. (Even if you have links specified, you'd still want the data: {} resource linkage present so Ember Data can cross-reference the included models).

@samselikoff: This description say's only if "links are present...", but this part is missing in `_maybeAddRelationshipsToResourceObjectForModel` (c297bb9). This leads to missing relationship data if links are not present and relationship is not included. But the relationship IDs in `data: {}` are still required to allow individual async loading of the relationship by ID.

**Comment by aalimovs** [comment]
I'm lost with this one. Using 0.3.4 with 2.15. 

I've got one-to-many relationship, e.g. `user` and `comments`, and I have nested API endpoint serving `/users/{id}/comments`. By default, comments are loaded asynchronously, `comments: DS.hasMany('comment', { async: true })`.

I've created a serializer for `user`:

```javascript
links(user) {
    return {
        'comments': {
            related: `/users/${user.id}/comments`
        }
    };
}
```

When retrieving a user, mirage returns the links, but also an empty data array:

```javascript
...
relationships: {
    comments: {
        data: [],
        links: {
            related: '/users/5/comments'
        }
    }
}
```

Ember picks up the empty array and considers there are no comments when doing `user.get('comments')`. I've created a `user` serializer in Ember that deletes `relationships.comments.data` and Ember does a network request as expected.

How do I make mirage not to return `data` key at all?
